pkg/sdn/plugin/controller.go

```diff
@@ -330,6 +337,155 @@ func (plugin *OsdnNode) SetupSDN(localSubnetCIDR, clusterNetworkCIDR, servicesNe
 	return true, nil
 }
 
+func (plugin *OsdnNode) SetupEgressNetworkPolicy() error {
+	policies, err := plugin.registry.GetEgressNetworkPolicies()
+	if err != nil {
+		return fmt.Errorf("Could not get EgressNetworkPolicies: %s", err)
+	}
+
+	for _, policy := range policies {
+		vnid, err := plugin.vnids.GetVNID(policy.Namespace)
+		if err != nil {
+			glog.Warningf("Could not find namespace %q for EgressNetworkPolicy: %v", err)
+			continue
+		}
+		plugin.egressPolicies[vnid] = append(plugin.egressPolicies[vnid], &policy)
+	}
+
+	for vnid := range plugin.egressPolicies {
+		err := plugin.updateEgressNetworkPolicy(vnid)
+		if err != nil {
+			return err
+		}
+	}
+
+	go utilwait.Forever(plugin.watchEgressNetworkPolicies, 0)
+	return nil
+}
+
+func (plugin *OsdnNode) watchEgressNetworkPolicies() {
+	eventQueue := plugin.registry.RunEventQueue(EgressNetworkPolicies)
+
+	for {
+		eventType, obj, err := eventQueue.Pop()
+		if err != nil {
+			utilruntime.HandleError(fmt.Errorf("EventQueue failed for EgressNetworkPolicy: %v", err))
+			return
+		}
+		policy := obj.(*osapi.EgressNetworkPolicy)
+
+		vnid, err := plugin.vnids.GetVNID(policy.Namespace)
+		if err != nil {
+			glog.Warningf("Could not find namespace %q for EgressNetworkPolicy: %v", err)
```

Comment:
glog.Warningf("Could not find netid for namespace %q: %v", policy.Namespace, err)